File: soyeong/programmers-2/올바른-괄호.py
def solution(s):
    answer = False
    stack = []

    # 짝 판별을 위해 꼭 다 저장해야한다는 내 편견을 부순 문제

    for i in range(len(s)):
        if s[i] == '(':
            stack.append(s[i])
        else:  # )을 만나면 바로 (를 삭제하여 남는걸 보는 방식
            if len(stack) == 0:
                return False  # stack이 비어있는 상태에서 )가 있다는 건 절대 짝이 불가함으로 False
            stack.pop()

    if len(stack) == 0:
        answer = True

    return answer

File: kyulee/programmers-2/올바른 괄호.java

public class 올바른 괄호 {

    class Solution {
        boolean solution(String s) {
            boolean answer = true;

            while (s.contains("()")) {
                s = s.replace("()", "");
            }

            return s.isEmpty();
        }
    }

    public static void main(String[] args) {
        String test = "()()";
        boolean result = solution(test);
        System.out.println(result);

    }
}